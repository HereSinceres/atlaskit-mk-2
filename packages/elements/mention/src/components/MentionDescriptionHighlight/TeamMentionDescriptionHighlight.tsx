import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { DescriptionHighlightProps } from './types';
import { DescriptionHighlightStyle } from './styles';

export default class TeamMentionDescriptionHighlight extends React.PureComponent<
  DescriptionHighlightProps,
  {}
> {
  renderTeamInformation(memberCount?: number, includesYou?: boolean) {
    // TEAMS-333 This same logic is implemented in User Picker as well. https://product-fabric.atlassian.net/browse/TEAMS-333
    // will extract this to a common place

    // if Member count is missing, do not show the byline, regardless of the availability of includesYou
    if (typeof memberCount !== 'number') {
      return null;
    }

    return (
      <DescriptionHighlightStyle>
        <FormattedMessage
          {...(memberCount > 50
            ? messages.plus50Members
            : messages.memberCount)}
          values={{ count: memberCount, includes: includesYou }}
        />
      </DescriptionHighlightStyle>
    );
  }

  render() {
    const { context } = this.props.mention;

    const includesYou = context && context.includesYou;
    const memberCount = context && context.memberCount;

    return this.renderTeamInformation(memberCount, includesYou);
  }
}

const messages = {
  memberCount: {
    id: 'fabric.elements.mentions.team.member.count',
    defaultMessage:
      '{count} {count, plural, one {member} other {members}}{includes, select, true {, including you} other {}}',
    description:
      'Number of members in the team and whether it includes the current user',
  },
  plus50Members: {
    id: 'fabric.elements.mentions.team.member.50plus',
    defaultMessage:
      '50+ members{includes, select, true {, including you} other {}}',
    description:
      'Number of members in a team exceeds 50 and whether it includes the current user',
  },
};
