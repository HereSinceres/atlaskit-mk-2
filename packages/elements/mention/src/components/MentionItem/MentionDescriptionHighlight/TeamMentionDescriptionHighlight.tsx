import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { Props } from './types';
import { TeamInformationStyle } from './styles';

function renderTeamInformation(memberCount?: number, includesYou?: boolean) {
  // todo - refactor with TeamOption ?
  // if Member count is missing, do not show the byline, regardless of the availability of includesYou
  if (memberCount === null || typeof memberCount === 'undefined') {
    return undefined;
  }
  return (
    <TeamInformationStyle>
      <FormattedMessage
        {...(memberCount > 50 ? messages.plus50Members : messages.memberCount)}
        values={{ count: memberCount, includes: includesYou }}
      />
    </TeamInformationStyle>
  );
}

export default class TeamMentionDescriptionHighlight extends React.PureComponent<
  Props,
  {}
> {
  render() {
    const { context } = this.props.mention;

    const includesYou = context && context.includesYou;
    const memberCount = context && context.memberCount;

    return renderTeamInformation(memberCount, includesYou);
  }
}

const messages = {
  memberCount: {
    id: 'fabric.elements.mentions.team.member.count',
    defaultMessage:
      '{count} {count, plural, one {member} other {members}}{includes, select, true {, including you} other {}}',
    description:
      'Number of members in the team and whether it includes the current user',
  },
  plus50Members: {
    id: 'fabric.elements.mentions.team.member.50plus',
    defaultMessage:
      '50+ members{includes, select, true {, including you} other {}}',
    description:
      'Number of members in a team exceeds 50 and whether it includes the current user',
  },
};
