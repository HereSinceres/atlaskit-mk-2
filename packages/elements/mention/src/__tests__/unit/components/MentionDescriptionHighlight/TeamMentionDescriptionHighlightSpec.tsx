import { render } from 'enzyme';
import * as React from 'react';
import TeamMentionDescriptionHighlight from '../../../../components/MentionDescriptionHighlight';
import { teamMention } from './_commonData';

const shallowRender = () =>
  render(<TeamMentionDescriptionHighlight mention={teamMention} />);

describe('Team mention description', () => {
  it('should render Team Mention description component', () => {
    const component = shallowRender();
    expect(component).toMatchSnapshot();
  });
});
