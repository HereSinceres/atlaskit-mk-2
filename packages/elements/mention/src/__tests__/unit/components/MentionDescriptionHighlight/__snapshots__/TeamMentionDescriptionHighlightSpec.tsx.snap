// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team mention description should render Team Mention description component 1`] = `
.c0 {
  color: #7A869A;
  font-size: 12px;
  margin-top: 2px;
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<span
  class="c0"
>
  <span>
    5 members, including you
  </span>
</span>
`;
